Make.inc

```diff
@@ -496,6 +494,8 @@ else
 ifneq ($(XC_HOST),)
 XC_HOST := $(ARCH)$(shell echo $(XC_HOST) | sed "s/[^-]*\(.*\)$$/\1/")
 MARCH := $(subst _,-,$(ARCH))
+else # insert ARCH into HOST
+XC_HOST := $(ARCH)$(shell echo $(BUILD_MACHINE) | sed "s/[^-]*\(.*\)$$/\1/")
```

Comment:
if you explicitly set `ARCH`, then it assumes you meant to try to do a cross-compile. without this, libunwind was getting confused by the mismatch between `--host` and `CC=gcc -m32`.